geopolars/src/geoseries.rs

```diff
@@ -624,6 +629,25 @@ impl GeoSeries for Series {
         }
     }
 
+    #[cfg(feature = "proj")]
+    fn to_crs(&self, from: &str, to: &str) -> Result<Series> {
+        use proj::{Proj, Transform};
+
+        let proj = Proj::new_known_crs(from, to, None).unwrap();
```

Comment:
This seems good for just now but it might be good to be able to support known projections and proj string definitions.

 We could perhaps implement an enum type 

```rust
enum ProjType {
    Known(string),
    Definition(string)
}
```

and implement `From<String> for ProjType ` in which we check to see if the string is of format "Authority:Code" in which case we return the Known variant and otherwise the Definition variant.

That would allow us to handle both cases while still allowing the user to pass in a simple string. Or we could even do without the enum and just use a regex to inspect the string and decide that way?

Reply:
I'm not sure I follow. Is your critique coming from the `known_crs` name?

This is my first time using the proj crate, but it looks like there are two ways to create a `Proj` object:
- Via [`Proj::new`](https://docs.rs/proj/latest/proj/struct.Proj.html#method.new), which uses a PROJ string definition like `"+proj=merc +lat_ts=56.5 +ellps=GRS80"` (and only takes that one argument)
- Via [`Proj::new_known_crs`](https://docs.rs/proj/latest/proj/struct.Proj.html#method.new_known_crs), which takes _both_ a `from` and a `to` argument, and takes a variety of string inputs:
    - an `"AUTHORITY:CODE"`, like `"EPSG:25832"`.
    - a PROJ string, like `"+proj=longlat +datum=WGS84"`
    - the name of a CRS as found in the PROJ database, e.g `"WGS84"`, `"NAD27"`, etc.
    - more generally, any string accepted by [new()](https://docs.rs/proj/latest/proj/struct.Proj.html#method.new)

It seems like here we'd always want the latter case? Where we want the high-level API to convert from one CRS to another?

The current behavior offloads the responsibility for passing in a valid PROJ CRS argument to the caller. I really don't want to do something like regex, because PROJ will do its own validation anyways. We should just be better about error handling (i.e. not have `unwrap`s like here)